flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -177,6 +177,56 @@ SqlDescribeDatabase SqlDescribeDatabase() :
 
 }
 
+SqlAlterFunction SqlAlterFunction() :
+{
+    SqlIdentifier functionName = null;
+    SqlCharStringLiteral functionClassName = null;
+    SqlCharStringLiteral functionLanguage = null;
+    SqlParserPos startPos;
+    boolean ifExists = false;
+    boolean isSystemFunction = false;
+}
+{
+    <ALTER>
+    <TEMPORARY>
+    (
+        <SYSTEM>   { isSystemFunction = true; }
+    |
+        {isSystemFunction = false; }
+    )
+    <FUNCTION> { startPos = getPos(); }
+    (
+        <IF> <EXISTS> { ifExists = true; }
+    |
+        { ifExists = false; }
+    )
+    functionName = CompoundIdentifier()
+    [ <AS> <QUOTED_STRING> {
+        String p = SqlParserUtil.parseString(token.image);
+        functionClassName = SqlLiteral.createCharString(p, getPos());
+    }]
+    [ <LANGUAGE> <QUOTED_STRING> {
```

Comment:
Not sure if what's our decision on language types, but have a quoted string for language seems suggestion user can put anything. One alternative is define constant such as SQL, PYTHON, JAVA, etc.